### ISSUE TITLE
[2.0.1] Deep Acl inheritance order issue

### ISSUE THREAD

**Issue by tdely** [phalcon/cphalcon#10294]
$acl->addInherit() will only consider the current rules of inherited roles, consider the following:

```php
$acl = new Phalcon\Acl\Adapter\Memory(); 
$acl->setDefaultAction(Phalcon\Acl::DENY);

$acl->addRole(new Phalcon\Acl\Role('Jane'));
$acl->addRole(new Phalcon\Acl\Role('Admin'));
$acl->addRole(new Phalcon\Acl\Role('User'));

$acl->addResource('Contact', ['ping', 'info', 'getAll']);

$acl->allow('Admin', 'Contact', 'ping');
$acl->allow('User', 'Contact', 'getAll');
$acl->allow('Jane', 'Contact', 'info');

$acl->addInherit('Jane', 'Admin');
$acl->addInherit('Admin', 'User');
```
Jane has access to Contact::ping and Contact::info

```php
$acl->addInherit('Admin', 'User');
$acl->addInherit('Jane', 'Admin');
```
Jane instead has access to Contact::ping, Contact::info and Contact::getAll

**Comment by Green-Cat** [comment]
May be related to #10209

**Comment by Green-Cat** [comment]
@andresgutierrez Fixed in #10209
